src/shared/approval-status/use-server-date-time-as-local.test.js

```diff
@@ -0,0 +1,126 @@
+     */
+    it('Converts correctly if on server time', () => {
+        useConfig.mockReturnValueOnce({
+            systemInfo: { serverTimeZoneId: 'Etc/UTC' },
+        })
+        const dateStr = '2021-08-31 16:00:00'
+        const localDateTime = new Date(dateStr)
+        const converted = useServerDateTimeAsLocal(dateStr)
+        expect(converted.getHours()).toEqual(localDateTime.getHours())
+    })
+
+    describe('timezone IDs unadjusted for daylight saving times', () => {
+        it('Converts correctly if ahead of server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Etc/GMT-9' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            expect(converted.getHours()).toEqual(localDateTime.getHours() - 9)
+        })
+
+        it('Converts correctly if behind on server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Etc/GMT+4' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            expect(converted.getHours()).toEqual(localDateTime.getHours() + 4)
+        })
+    })
+
+    describe('timezone IDs adjusted for daylight saving times', () => {
+        it('Converts correctly if ahead of server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'Europe/Amsterdam' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            const hourOffset = localDateTime.getHours() - converted.getHours()
+            expect([1, 2]).toContain(hourOffset)
+        })
+
+        it('Converts correctly if behind on server time', () => {
+            useConfig.mockReturnValueOnce({
+                systemInfo: { serverTimeZoneId: 'America/Chihuahua' },
+            })
+            const dateStr = '2021-08-31 16:00:00'
+            const localDateTime = new Date(dateStr)
+            const converted = useServerDateTimeAsLocal(dateStr)
+            const hourOffset = localDateTime.getHours() - converted.getHours()
+            expect([-7, -6]).toContain(hourOffset)
+        })
+    })
+
+    it('will throw an error if the server timezone ID is invalid', () => {
+        useConfig.mockReturnValueOnce({
+            systemInfo: { serverTimeZoneId: 'I_DO_NOT_EXISTS' },
+        })
+        expect(() =>
+            useServerDateTimeAsLocal('2021-08-31 16:00:00')
+        ).toThrowError('Invalid time zone specified: I_DO_NOT_EXISTS')
+    })
+
+    describe('day transitions', () => {
+        it('only changes hours when in the middle of the day', () => {
```

Comment:
```suggestion
        it('changes none but hours when in the middle of the day', () => {
```

I interpret the current scenario string as "it would only change hours if in the middle of the day, but wouldn't change hours when not in the middle of the day". But I guess a native speaker should confirm